### PR TITLE
202311 nulls

### PR DESCRIPTION
**Pull request by petermarshallio**
A notebook on transforming rows of incoming data to create NULL, with some examples of situations when this might be necessary. Followed by some example scalar function operations, and how NULL is addressed in aggregations. Also includes a couple of examples of NULL being used with arrays.

### CODE REVIEW

File: notebooks/02-ingestion/08-generating-and-working-with-nulls.ipynb [hunk 1]
```diff
@@ -0,0 +1,812 @@
...
+    "  ~\n",
+    "  ~ Unless required by applicable law or agreed to in writing,\n",
+    "  ~ software distributed under the License is distributed on an\n",
+    "  ~ \"AS IS\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY\n",
+    "  ~ KIND, either express or implied.  See the License for the\n",
+    "  ~ specific language governing permissions and limitations\n",
+    "  ~ under the License.\n",
+    "  -->\n",
+    "\n",
+    "In this notebook, you will see examples of various functions and operators used with NULL values, generating some NULL data through different SQL-based ingestions, and then working with them using a variety of functions."
```
Comment (techdocsmith) [inline notebooks/02-ingestion/08-generating-and-working-with-nulls.ipynb:28]: Might want a little more context here. Why is the behavior of NULLS important.
Also I feel like it's important with comparisons like inequality. NULLS aren't returned in inequality according to the SQL standard.

**Review by techdocsmith — APPROVED** [review]
This version is a great start for a v1. It is a little long, but good detail.

File: notebooks/02-ingestion/08-generating-and-working-with-nulls.ipynb (hunk 1 above)
Comment (petermarshallio) [reply notebooks/02-ingestion/08-generating-and-working-with-nulls.ipynb:28]: Coolio - I'll add a little bit of an intro :)

**Comment by petermarshallio** [comment]
> This version is a great start for a v1. It is a little long, but good detail.

Yeah I feel like maybe there's an option here to just pull some of it alongside other notebooks. Noted for a v2 😆

**Pull request merged**